ambry-frontend/src/main/java/com/github/ambry/frontend/GetBlobHandler.java

```diff
@@ -162,21 +167,11 @@ private Callback<Void> securityPostProcessRequestCallback(BlobId blobId) {
           RestRequestMetrics restRequestMetrics = metricsGroup.getRestRequestMetrics(restRequest.isSslUsed(), true);
           restRequest.getMetricsTracker().injectMetrics(restRequestMetrics);
         }
-        if (subResource == null) {
+        if (subResource == SubResource.Replicas) {
+          finalCallback.onCompletion(getReplicasHandler.getReplicas(blobId.getID(), restResponseChannel), null);
+        } else {
```

Comment:
Please don't refactor. Make a separate PR if you want to refactor.

Reply:
Actually after a second thought, I decided not to refactor this code. Instead I switch back to the original coding style so in the future when people add more subResource, they will quickly get this code need to be updated.